pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -68,6 +68,11 @@ const (
 	errorStatus  = "ERROR"
 
 	ServiceAnnotationLoadBalancerFloatingNetworkId = "loadbalancer.openstack.org/floating-network-id"
+
+	// ServiceAnnotationLoadBalancerInternal is the annotation used on the service
+	// to indicate that we want an internal loadbalancer service.
+	// If the value of ServiceAnnotationLoadBalancerInternal is false, it indicates that we want an external loadbalancer service. Default to true.
+	ServiceAnnotationLoadBalancerInternal = "service.beta.kubernetes.io/openstack-load-balancer-internal"
```

Comment:
ServiceAnnotationLoadBalancerFloatingNetworkId is the annotation used on floating-network-id, the same as 'floating-network-id' which from cloud provider config file.
ServiceAnnotationLoadBalancerInternal is the annotation used to distinguish between internal LoadBalancerIP and external LoadBalancerIP.
If it is true, the service.Spec.LoadBalancerIP is a vip of neutron LB.
If it is false,  the service.Spec.LoadBalancerIP is a floatingip of neutron LB's vip.

Currently openstack cloud provider just support for specifying internal LoadBalancerIP(vip). I think that is unreasonable, actually user just use a floating ip to access a LoadBalancer service. So it is necessary to spport for specifying external LoadBalancerIP(floatingip).